lib/api/src/rc_switch_user.cpp

```diff
@@ -11,25 +11,22 @@
 auto rc_switch_user(RcComm* _comm, const char* _username, const char* _zone) -> int
 {
     // NOLINTNEXTLINE(readability-implicit-bool-conversion)
-    if (!_username || !_zone) {
+    if (!_comm || !_username || !_zone) {
         return SYS_INVALID_INPUT_PARAM;
     }
 
     auto& client = _comm->clientUser;
 
-    // Attempting to verify that the user is already the target user cannot be done without
-    // going over the wire. That's because the client cannot determine the user type of the
-    // target user. For that reason, this implementation always results in a network request.
+    if (std::strcmp(_username, client.userName) == 0 && std::strcmp(_zone, client.rodsZone) == 0) {
```

Comment:
please add a comment about why this is allowed / good / efficient.

Reply:
Done